tests/pytests/unit/roster/test_dir.py

```diff
@@ -0,0 +1,181 @@
+"""
+Test the directory roster.
+"""
+
+
+import logging
+
+import pytest
+import salt.config
+import salt.loader
+import salt.roster.dir as dir_
+
+
+@pytest.fixture
+def roster_domain():
+    return "test.roster.domain"
+
+
+@pytest.fixture
+def expected(roster_domain):
+    return {
+        "basic": {
+            "test1_us-east-2_test_basic": {
+                "host": "127.0.0.2",
+                "port": 22,
+                "sudo": True,
+                "user": "scoundrel",
+            }
+        },
+        "domain": {
+            "test1_us-east-2_test_domain": {
+                "host": "test1_us-east-2_test_domain." + roster_domain,
+                "port": 2222,
+                "user": "george",
+            }
+        },
+        "empty": {
+            "test1_us-east-2_test_empty": {
+                "host": "test1_us-east-2_test_empty." + roster_domain,
+            }
+        },
+    }
+
+
+@pytest.fixture
+def create_roster_files(tmp_path):
+    badfile_contents = """
+    #!jinja|yaml
+    host: 127.0.0.2
+    port: 22
+    THIS FILE IS NOT WELL FORMED YAML
+    sudo: true
+    user: scoundrel
+    """
+
+    basic_contents = """
+    #!jinja|yaml
+    host: 127.0.0.2
+    port: 22
+    sudo: true
+    user: scoundrel
+    """
+
+    domain_contents = """
+    #!jinja|yaml
+    port: 2222
+    user: george
+    """
+
+    empty_contents = """
+    """
+
+    with pytest.helpers.temp_file(
+        "test1_us-east-2_test_badfile", badfile_contents, directory=tmp_path
+    ), pytest.helpers.temp_file(
+        "test1_us-east-2_test_basic", basic_contents, directory=tmp_path
+    ), pytest.helpers.temp_file(
+        "test1_us-east-2_test_domain", domain_contents, directory=tmp_path
+    ), pytest.helpers.temp_file(
+        "test1_us-east-2_test_empty", empty_contents, directory=tmp_path
+    ):
+        yield
+
+
+@pytest.fixture
+def configure_loader_modules(roster_domain, salt_master_factory, tmp_path):
+
+    opts = salt_master_factory.config.copy()
+    utils = salt.loader.utils(opts, whitelist=["json", "stringutils", "roster_matcher"])
+    runner = salt.loader.runner(opts, utils=utils, whitelist=["salt"])
+    return {
+        dir_: {
+            "__opts__": {
+                "extension_modules": "",
+                "optimization_order": [0, 1, 2],
+                "renderer": "jinja|yaml",
+                "renderer_blacklist": [],
+                "renderer_whitelist": [],
+                "roster_dir": tmp_path,
```

Comment:
```suggestion
                "roster_dir": str(tmp_path),
```
Because of Debian 9 running Py3.5?
Unless tests don't fail be side of this....